clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -268,16 +268,7 @@ void CodeGenModule::createCUDARuntime() {
 }
 
 void CodeGenModule::createSYCLRuntime() {
-  switch (getTriple().getArch()) {
-  case llvm::Triple::spir:
-  case llvm::Triple::spir64:
-  case llvm::Triple::nvptx:
-  case llvm::Triple::nvptx64:
-    SYCLRuntime.reset(new CGSYCLRuntime(*this));
-    break;
-  default:
-    llvm_unreachable("unsupported target for SYCL");
-  }
+  SYCLRuntime.reset(new CGSYCLRuntime(*this));
```

Comment:
*thank you for getting rid of that stupid `llvm_unreachable`!*

This closes: https://github.com/intel/llvm/issues/3622